pkg/storage/formattedstore/formattedstore.go

```diff
@@ -122,14 +168,31 @@ func (s *formatStore) Get(k string) ([]byte, error) {
 		return nil, fmt.Errorf(failParseFormattedValue, err)
 	}
 
+	// put data in cache to retrieve it next time
+	if s.cacheStore != nil {
+		if err := s.cacheStore.Put(k, value); err != nil {
+			return nil, err
+		}
+	}
+
 	return value, nil
+
 }
 
 // The "don't filter" switch is here since it's possible for the underlying storage to have already done
 // the filtering, in which case it would be redundant to do it here too.
 // TODO (#2315) While doing the storage interface rework, see if the skipIteratorFiltering switch is still
 //  needed.
 func (s *formatStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	if s.cacheStore != nil {
+		r := s.cacheStore.Iterator(startKey, endKey)
+		if r.Error() == nil {
+			return r
```

Comment:
Currently the cacheStore can be any type of provider. I know it would be silly, but if you made the cacheStore another EDV REST provider, then this will not work correctly. The EDV REST store's iterator method ignored the startKey and endKey and always returns all document.

(Even if you ignore this weird edge case, you should have a comment with a warning or something like that)

Reply:
@DRK3 this interface will change correct ?

Reply:
@fqutishat Eventually, yeah. I want to get rid of that "skipIteratorFiltering" check (see the TODO above this method).